syntax = "proto3";

package proto;

service CoreServices {
  rpc Ping(PingRequest) returns (PingReply);
  rpc Sum(SumRequest) returns (SumReply);
  rpc StreamNumbers(StreamRequest) returns (stream StreamReply);
}

message PingRequest { string name = 1; }
message PingReply { string message = 1; }

message SumRequest { int32 a = 1; int32 b = 2; }
message SumReply { int32 result = 1; }

message StreamRequest { int32 max = 1; }
message StreamReply { int32 value = 1; }
